packages/web-components/src/wallet-connection/watchWallet.js

```diff
@@ -125,9 +128,33 @@ export const watchWallet = (chainStorageWatcher, address, rpc) => {
         updatePurses(brandToPurse);
       };
 
-      const watchBank = async () => {
-        const balances = await queryBankBalances(address, rpc);
-        bank = balances;
+      /** @type {Tendermint34Client} */
+      let tendermintClient;
+      const watchBank = async (numRetries = 0) => {
+        await null;
+        try {
+          tendermintClient ||= await Tendermint34Client.connect(rpc);
+        } catch (e) {
+          console.error('Error connecting to rpc endpoint', rpc);
+          if (numRetries > 1) {
+            onRpcError && onRpcError(e);
+          } else {
+            setTimeout(() => watchBank(numRetries + 1), RETRY_INTERVAL_MS);
+            return;
+          }
+        }
+
+        try {
+          bank = await queryBankBalances(address, tendermintClient);
+        } catch (e) {
+          console.error('Error querying bank balances for address', address);
```

Comment:
The catch here is the mostly same so I take it you made a separate try in order to have a distinct console message. Could you handle that instead by inspecting `e`?

If you want to maintain the separate try, please DRY the rest of the catch.

Reply:
I think it's not too important or helpful to have a separate console message, wasn't really thinking it through well. I agree it's cleaner to just put them in the same `try`, so just did that.